Cython/Compiler/ExprNodes.py

```diff
@@ -849,6 +845,47 @@ def free_subexpr_temps(self, code):
     def generate_function_definitions(self, env, code):
         pass
 
+    # ----Generation of small bits of reference counting --
+
+    #@property
+    #def refcounting_type(self):
+        # FIXME this might just be "ctype"?
+    #    type = self.type
+    #    if self.is_temp and self.type.is_pyobject:
+            # when a temp object is created it's always just
+            # created as a py_object_type and not a
+            # derived type
+    #        type = PyrexTypes.py_object_type
+    #    return type
+
+    def generate_decref_set(self, code, rhs, handle_null=False):
+        if not (handle_null and self.cf_is_null):
```

Comment:
This replicates the previous behaviour in the place it's used https://github.com/cython/cython/blob/4dd7c18de76cb5c90a5168fba35990600307e642/Cython/Compiler/ExprNodes.py#L2358-L2367.

However, since it's only used once and the behaviour doesn't make sense for `generate_decref_set` I think I'll remove the `handle_null` option from `generate_decref_set` and just return the place where it's used to doing it explicitly.